### PR TITLE
Add org.wireshark.qt

### PR DESCRIPTION
**Pull request by muelli**
This builds wireshark. It may be useful to have a newer Wireshark build
to analyse protocols which are not yet in the distribution's wireshark
version.

### CODE REVIEW

**Comment by muelli** [comment]
I have no idea.
I just needed a newer version of Wireshark and put together a flatpaked version.

**Comment by TingPing** [comment]
Well you'll want to package the Qt version.

**Comment by barthalion** [comment]
Has Qt frontend reached feature parity with GTK one? Last time I checked it missed some features, but that was years ago.

**Comment by TingPing** [comment]
I believe so. Looks like its already been deleted in master so its definitely a dead end.

**Comment by eszlari** [comment]
How do you fix the `dumpcap` permission problem:
https://wiki.wireshark.org/CaptureSetup/CapturePrivileges

**Comment by TingPing** [comment]
Probably simply can't.

File: org.wireshark.qt.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    cleanup:
+        - "/include"
+        - "/share"
+    sources:
+        - type: archive
+          url: https://www.tcpdump.org/release/libpcap-1.9.0.tar.gz
+          sha256: 2edb88808e5913fdaa8e9c1fcaf272e19b2485338742b5074b9fe44d68f37019
+
+
+  - name: wireshark
```
Comment (TingPing) [inline org.wireshark.qt.yml]: Looks like new versions are going purely `cmake` also, might as well follow.

Comment (muelli) [reply org.wireshark.qt.yml]: I don't quite follow. Do you think we should add `buildsystem: cmake` now?

Comment (TingPing) [reply org.wireshark.qt.yml]: Well `cmake-ninja` but yes. No point in using tooling that is already deleted.

**Comment by nedrichards** [comment]
bot, build org.wireshark.qt
testbot, build org.wireshark.qt

**Comment by barthalion** [comment]
I suspect that upstream still prefers AppImage – but can you get in touch with them about flatpak package?

testbot, build org.wireshark.qt

**Comment by eszlari** [comment]
Apart from that `dumpcap` problem, I would connect upstream about the `app-id`. I'm not sure, they would be happy with `org.wireshark.qt`.

**Comment by rknall** [comment]
Just a question, are there any rules for such names? I am one of the core devs of wireshark. Personally I do not like the idea of org.wireshark.qt. The dumpcap is not really an issue for me atm.

File: org.wireshark.gtk.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+app-id: org.wireshark.gtk
```
Comment (crondaemon) [inline org.wireshark.gtk.yml]: The GTK ui doesn't exist anymore. You can delete this file.

File: org.wireshark.qt.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    config-opts:
+        - --with-qt
```
Comment (crondaemon) [inline org.wireshark.qt.yml]: This should not be needed. It's the default.

**Comment by Lekensteyn** [comment]
(Another upstream developer chiming in, speaking for myself. Not familiar with Flatpak configuration and Arch Linux/KDE user.)

The name org.wireshark.qt indeed sounds odd. Based on some other examples (org.vlc.VLC, org.gimp.GIMP, basically [vendor].[application name]), `org.wireshark.Wireshark` seems more appropriate. Initially I also though about `org.wireshark.gui`, but since this package presumably include CLI tools (like tshark, capinfos, etc.) as well, I don't think that would be an appropriate name either.

Others already mentioned it, but for completeness:
- GTK+ support is removed in master and no longer supported. It was already deprecated in 2.6 and earlier, so new packages (e.g. Flatpak) should definitely not use it. There are known bugs that we won't fix. Publishing it might confuse users. The 3.0 release should be "soon", see https://wiki.wireshark.org/Development/Roadmap (slightly behind schedule)
- autotools support has also been removed in master. Make sure to build with CMake to make further updates easier.

Some other feedback:
Why set `NPM_CONFIG_LOGLEVEL`? That seems like an option for nodejs and not relevant for Wireshark?

There are quite some other required and optional dependencies. I only see libpcap declared, but there are others. See also https://wiki.wireshark.org/Development/Support_library_version_tracking
From top of my head, required:
- libgcrypt. Newer versions are recommended as they enable features such as TLS 1.3 decryption.
- GLib
- (build process, not runtime:) Python 3.4 or newer.
Optional:
- GnuTLS. 3.2 or newer will be required for Wireshark 3.0, used to enable RSA decryption.
- Lua 5.2: for Lua plugin support.
- libmaxmindb: for GeoIP support.
- nghttp2: for HTTP/2 dissection support.
- libssh: to enable a ssh client for remote capture support.
- ... and others. Run CMake and/or see CMakeOptions.txt for some more dependencies.

There is also a test suite. In 2.6 and before you can run test/test.sh, starting with 3.0 there is a Python test suite in test/test.py, but you can also run `pytest` (see https://www.wireshark.org/docs/wsdg_html_chunked/ChapterTests.html, "Listing And Running Tests (pytest)". Is there a way to run this?

File: org.wireshark.qt.yml (hunk 3 above)
Comment (crondaemon) [reply org.wireshark.qt.yml]: Sorry, my mistake. This is an autotools option, and that build system doesn't exist anymore.

**Comment by TingPing** [comment]
> at some point we might be provide an AppImage version, so a Flatpak version might not be necessary. See bugs.wireshark.org/bugzilla/show_bug.cgi?id=14464

Well Flatpak has strong guarantees of portability while AppImage doesn't (it becomes your job to ensure you don't use a single host lib (you will). But Flatpak introduces the complexity of a sandbox so you can't do things that require root access anymore.

**Comment by Lekensteyn** [comment]
@muelli What does `wireshark --version` show for this build?

**Comment by muelli** [comment]
```
flatpak --user run --command=bash org.wireshark.qt/x86_64/master 
bash-4.4$ wireshark --version
Wireshark 2.6.6 (v2.6.6)

Copyright 1998-2019 Gerald Combs <gerald@wireshark.org> and contributors.
License GPLv2+: GNU GPL version 2 or later <http://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
This is free software; see the source for copying conditions. There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiled (64-bit) with Qt 5.12.0, with libpcap, with POSIX capabilities (Linux),
without libnl, with GLib 2.58.1, with zlib 1.2.11, without SMI, without c-ares,
without Lua, with GnuTLS 3.6.5, with Gcrypt 1.8.3, with MIT Kerberos, without
MaxMind DB resolver, without nghttp2, without LZ4, without Snappy, with libxml2
2.9.8, with QtMultimedia, without SBC, without SpanDSP, without bcg729.

Running on Linux 4.15.0-38-generic, with Intel(R) Core(TM) i5-7200U CPU @
2.50GHz (with SSE4.2), with 15574 MB of physical memory, with locale C, with
libpcap version 1.9.0-PRE-GIT (with TPACKET_V3), with GnuTLS 3.6.5, with Gcrypt
1.8.3, with zlib 1.2.11, binary plugins supported (0 loaded).

Built using gcc 8.2.0.
bash-4.4$ 
```

**Comment by muelli** [comment]
and it crashes when trying to open a file...

```

(wireshark:2): GLib-GIO-CRITICAL **: 17:11:27.124: g_unix_is_system_fs_type: assertion 'fs_type != NULL && *fs_type != '\0'' failed

(wireshark:2): GLib-GIO-CRITICAL **: 17:11:27.124: g_unix_is_system_device_path: assertion 'device_path != NULL && *device_path != '\0'' failed
```

that feature does not exist in the Gtk version.

**Comment by TingPing** [comment]
@muelli 	I believe this is a bug elsewhere: https://github.com/flatpak/flatpak/issues/2558

**Comment by nedrichards** [comment]
bot, build org.wireshark.qt
testbot, build org.wireshark.qt

**Comment by barthalion** [comment]
testbot, build og.wireshark.Wireshark

**Comment by muelli** [comment]
testbot, build org.wireshark.Wireshark

**Comment by TingPing** [comment]
Is it good to go? Doesn't the sandbox limit core functionality from working as discussed?

**Comment by barthalion** [comment]
It shouldn't have a problem to open local pcap file.

**Comment by barthalion** [comment]
build org.wireshark.Wireshark

**Comment by barthalion** [comment]
bot, build org.wireshark.Wireshark

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.wireshark.Wireshark

Thanks!

@Lekensteyn, @crondaemon, @rknall, I have invited you to collaborators in case you would like to help with maintenance/PR reviews.

**Comment by Lekensteyn** [comment]
Could you look into enabling missing features:

> Compiled (64-bit) with Qt 5.12.0, with libpcap, with POSIX capabilities (Linux),
> without libnl, with GLib 2.58.1, with zlib 1.2.11, without SMI, **without c-ares**,
> **without Lua**, with GnuTLS 3.6.5, with Gcrypt 1.8.3, with MIT Kerberos, without
> MaxMind DB resolver, **without nghttp2**, without LZ4, without Snappy, with libxml2
> 2.9.8, with QtMultimedia, without SBC, without SpanDSP, without bcg729.

Features:
- DNS name resolution for IP addresses: c-ares
- Lua plugins/scripting: lua
- Full HTTP/2 support: nghttp2
- GeoIP: libmaxmind (also needs database files for full functionality though)

**Comment by muelli** [comment]
follow me: https://github.com/flathub/org.wireshark.Wireshark/issues/1#issue-402647414

if you can provide the names and urls of the missing libraries, it should be easy to add those to the build.

**Pull request closed without merging**